### PR TITLE
Add tests for binary type and fix casting of default value

### PR DESCRIPTION
**Pull request by Tigrov**
There are two commits:
1. with stream_get_contents - to cast query result to `string` using `stream_get_contents()`
Due to `AbstractCommand::queryScalar()` cast it to `string`
https://github.com/yiisoft/db/blob/master/src/Command/AbstractCommand.php#LL460C8-L460C8
2. without stream_get_contents

How do you think?

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | ❌

### CODE REVIEW

**Review by terabytesoftw — APPROVED** [review]
For me it's ok to use `is_string()` && `str_starts_with,()`, i tested `yiisoft/cache-db`, which uses binary data for `Pgsql` and it works great.

File: src/ColumnSchema.php [hunk 1]
```diff
@@ -146,6 +149,11 @@ protected function phpTypecastValue(mixed $value): mixed
                     'f', 'false' => false,
                     default => (bool)$value,
                 };
+            case SchemaInterface::TYPE_BINARY:
+                if (is_string($value) && str_starts_with($value, '\\x')) {
+                    return hex2bin(substr($value, 2));
```
Comment (darkdef) [inline src/ColumnSchema.php:154]: Please cover tests this lines

Comment (Tigrov) [reply src/ColumnSchema.php]: Test case for a default value covers the code.
Moved the code to the place where default values are casting.

**Review by samdark — APPROVED** [review]
Please add a line for CHANGELOG. Thanks.

**Comment by samdark** [comment]
Thank you!

**Pull request merged**